### PR TITLE
Improve SimplicialComplex

### PR DESCRIPTION
**Pull request by alexej-jordan**
This branch includes a testset for `SimplicialComplex`, improved type handling for the constructor and the function `fundamental_group([::FPGroup, ] ::SimplicialComplex)`. The latter requires changes from the branch of same name in `Polymake.jl`.

### CODE REVIEW

File: src/Combinatorics/SimplicialComplexes.jl [hunk 1]
```diff
@@ -280,6 +286,38 @@ Multivariate Polynomial Ring in 6 variables a, b, c, d, ..., f over Integer Ring
...
+function fundamental_group(K::SimplicialComplex)
+    n, r = pm_object(K).FUNDAMENTAL_GROUP
+    F = free_group(n)
+    return fundamental_group(F, K)
+end
+
+@doc Markdown.doc"""
+    fundamental_group([F::FPGroup,] K::SimplicialComplex)
+
+Return the fundamental group of the abstract simplicial complex `K`, as a quotient of a given Group `F`.
```
Comment (fingolfin) [inline src/Combinatorics/SimplicialComplexes.jl]: Group should be lower case...
```suggestion
Return the fundamental group of the abstract simplicial complex `K`, as a quotient of a given group `F`.
```
But more importantly:
- how is `F` chosen if not given?
- what does `F` have to satisfy to be suitable? Should it perhaps be a free group of a certain rank? If it not a free group, and/or has a different rank, what is the result?!?

File: src/Combinatorics/SimplicialComplexes.jl [hunk 2]
```diff
@@ -280,6 +286,38 @@ Multivariate Polynomial Ring in 6 variables a, b, c, d, ..., f over Integer Ring
...
+"Z x Z"
+```
+"""
+function fundamental_group(F::FPGroup, K::SimplicialComplex)
+    n, r = pm_object(K).FUNDAMENTAL_GROUP
+    rvec = Vector{FPGroupElem}(undef, length(r))
+    for (i, relation) in enumerate(r)
+        relem = one(F)
+        for term in relation
+            relem *= F[first(term) + 1]^last(term)
```
Comment (fingolfin) [inline src/Combinatorics/SimplicialComplexes.jl:312]: So this suggests that a certain number of generators is required at least; and when `F` is not free, then essentially garbage is produced?

File: test/Combinatorics/SimplicialComplexes.jl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            
+            @test SC isa SimplicialComplex
+            @test f_vector(SC) == fv
+            @test betti_numbers(SC) == bn
+            
+        end
+        
+    end
+    
+end
```
Comment (fingolfin) [inline test/Combinatorics/SimplicialComplexes.jl]: Please always end files with newline

File: src/Combinatorics/SimplicialComplexes.jl (hunk 2 above)
Comment (micjoswig) [reply src/Combinatorics/SimplicialComplexes.jl:312]: Don't worry.  @alexej-jordan produced this code only for me, to guide my implementation (using fresh Polymake.jl features)  I will take care of the math.

File: test/Combinatorics/SimplicialComplexes.jl (hunk 3 above)
Comment (micjoswig) [reply test/Combinatorics/SimplicialComplexes.jl]: So. Please do not merge into master!  Instead I will merge it into my current branch,

File: test/Combinatorics/runtests.jl [hunk 4]
```diff
@@ -0,0 +1,3 @@
+using Oscar
+
+include("SimplicialComplexes.jl")
```
Comment (benlorenz) [inline test/Combinatorics/runtests.jl]: This file is not included in the main `test/runtests.jl` so the tests do not run.

**Comment by benlorenz** [comment]
@micjoswig Please be aware that the changes here require changes in libpolymake_julia and Polymake.jl to be merged and released before the tests can succeed in the Oscar.jl repo. And any adjustments to the code need to be done in this branch here as the branchname needs to match the other two branches for the tests in those repositories to succeed.

**Comment by micjoswig** [comment]
> And any adjustments to the code need to be done in this branch here as the branchname needs to match the other two branches for the tests in those repositories to succeed.

Too late.  Now I created a mess by merging and pushing into my own branch.  But please remove the version of the fundamental group with given `F` for now.  Only keep `fundamental_group` with `K` as the sole argument.

**Comment by benlorenz** [comment]
This PR now includes the version bump to Polymake.jl 0.7 (needed for the changes in polymake 4.6 and libpolymake_julia 0.7) and since I just released Polymake.jl 0.7 the test will hopefully run through now.

**Pull request merged**